clients/filesystem-fuse/src/config.rs

```diff
@@ -0,0 +1,324 @@
+);
+
+const FUSE_FS_TYPE: ConfigEntity<&'static str> = ConfigEntity::new(
+    FuseConfig::MODULE_NAME,
+    "fs_type",
+    "The type of the FUSE filesystem",
+    "memory",
+);
+
+const FUSE_CONFIG_PATH: ConfigEntity<&'static str> = ConfigEntity::new(
+    FuseConfig::MODULE_NAME,
+    "config_path",
+    "The path of the FUSE configuration file",
+    "/etc/gvfs/gvfs.toml",
+);
+
+const FILESYSTEM_BLOCK_SIZE: ConfigEntity<u32> = ConfigEntity::new(
+    FilesystemConfig::MODULE_NAME,
+    "block_size",
+    "The block size of the gvfs fuse filesystem",
+    4096,
+);
+
+const GRAVITINO_URL: ConfigEntity<&'static str> = ConfigEntity::new(
+    GravitinoConfig::MODULE_NAME,
+    "gravitino_url",
+    "The URL of the Gravitino server",
+    "http://localhost:8090",
+);
+
+const GRAVITINO_METALAKE: ConfigEntity<&'static str> = ConfigEntity::new(
+    GravitinoConfig::MODULE_NAME,
+    "metalake",
+    "The metalake of the Gravitino server",
+    "",
+);
+
+struct ConfigEntity<T: 'static> {
+    module: &'static str,
+    name: &'static str,
+    description: &'static str,
+    default: T,
+}
+
+impl<T> ConfigEntity<T> {
+    const fn new(
+        module: &'static str,
+        name: &'static str,
+        description: &'static str,
+        default: T,
+    ) -> Self {
+        ConfigEntity {
+            module: module,
+            name: name,
+            description: description,
+            default: default,
+        }
+    }
+}
+
+enum ConfigValue {
+    I32(ConfigEntity<i32>),
+    U32(ConfigEntity<u32>),
+    String(ConfigEntity<&'static str>),
+    Bool(ConfigEntity<bool>),
+    Float(ConfigEntity<f64>),
+}
+
+struct DefaultConfig {
+    configs: HashMap<String, ConfigValue>,
+}
+
+impl Default for DefaultConfig {
+    fn default() -> Self {
+        let mut configs = HashMap::new();
+
+        configs.insert(
+            Self::compose_key(FUSE_DEFAULT_FILE_MASK),
+            ConfigValue::U32(FUSE_DEFAULT_FILE_MASK),
+        );
+        configs.insert(
+            Self::compose_key(FUSE_DEFAULT_DIR_MASK),
+            ConfigValue::U32(FUSE_DEFAULT_DIR_MASK),
+        );
+        configs.insert(
+            Self::compose_key(FUSE_FS_TYPE),
+            ConfigValue::String(FUSE_FS_TYPE),
+        );
+        configs.insert(
+            Self::compose_key(FUSE_CONFIG_PATH),
+            ConfigValue::String(FUSE_CONFIG_PATH),
+        );
+        configs.insert(
+            Self::compose_key(GRAVITINO_URL),
+            ConfigValue::String(GRAVITINO_URL),
+        );
+        configs.insert(
+            Self::compose_key(GRAVITINO_METALAKE),
+            ConfigValue::String(GRAVITINO_METALAKE),
+        );
+        configs.insert(
+            Self::compose_key(FILESYSTEM_BLOCK_SIZE),
+            ConfigValue::U32(FILESYSTEM_BLOCK_SIZE),
+        );
+
+        DefaultConfig { configs }
+    }
+}
+
+impl DefaultConfig {
+    fn compose_key<T>(entity: ConfigEntity<T>) -> String {
+        format!("{}.{}", entity.module, entity.name)
```

Comment:
This implies that we have some constraints on the values of `entity.module`, `entity.name`, right?
For example, they cannot contain `.`.

Reply:
There is no such restriction.